registry/github_registry.go

```diff
@@ -38,6 +39,7 @@ type githubRegistry struct {
 	format         common.RegistryFormat
 	credentialName string
 	service        GithubRepositoryService
+	httpClient     *http.Client
```

Comment:
It only unmarshals JSON if you pass it an object to read. Otherwise, it just copies the response body to the writer.